sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/OperatorStateMetadata.scala

```diff
@@ -166,18 +169,28 @@ object OperatorStateMetadataUtils extends Logging {
           s"version=${operatorStateMetadata.version}")
     }
   }
+
+  def getLastOffsetBatch(session: SparkSession, checkpointLocation: String): Long = {
```

Comment:
Just thinking out loud here, so we are reading metadata from latest batch in the offsetlog. Suppose a query failed before committing the batch and write nothing to the metadata path, and then we start a metadata reader from it: could there be situation where a batchId exists in offsetLog but there is actually no metadata written at this batchId?

Reply:
This I am only using in the context of the state metadata source to assign the default batch id. But I guess its fair to say that this can happen. I think its probably better to just use last committed batch in this case. Will update accordingly

Reply:
Done